programs/bench.c

```diff
@@ -286,6 +286,7 @@ static int BMK_benchMem(const void* srcBuffer, size_t srcSize,
                     U32 nbLoops = 0;
                     ZSTD_CCtx_setParameter(ctx, ZSTD_p_nbThreads, g_nbThreads);
                     ZSTD_CCtx_setParameter(ctx, ZSTD_p_compressionLevel, cLevel);
+                    //ZSTD_CCtx_setPledgedSrcSize(ctx, blockTable[0].srcSize);  /* estimation, only to help cLevel conversion before applying advanced compression parameters */
```

Comment:
Some compiler won't like the `//` comments right?

Reply:
Yes, some old ones since `//` is not part of `C90`

Reply:
The choice here is between suppressing the line and enabling it.

Enabling the line should improve compression adaptation, notably for small files, when dictionary or advanced parameters are used.

But I'm concerned that this adaptation behaviour could be "specific" to benchmark mode, hence misleading.

The behavior should be replicated on the command line side.
As this is a different topic, I delayed the decision with the `//` comment.
It requires reading and potentially modifying `fileio.c`, so a bit more work.
It should be part of another patch.

If `//` is problematic, I can suppress the line in this patch.

Reply:
IIRC Appveyor tests will reject `//`, but I don't remember if it is MinGW or Visual.

Reply:
I believe it requires a much older compiler to fail the `//` comment test.

One way to fail it is to compile with `gcc` and `-std=c90 -pedantic`.
But in this case, a ton of other things fail too (typically, this setup does not support `unsigned long long` type).

Reply:
I must be misremembering the failure then.